terraform/deployments/govuk-publishing-platform/backends_origin.tf

```diff
@@ -0,0 +1,177 @@
+  extra_aliases                        = compact([local.is_default_workspace ? "publisher.${var.publishing_service_domain}" : null, "publisher.${aws_route53_zone.workspace_public.name}", "signon.${aws_route53_zone.workspace_public.name}"])
+  load_balancer_certificate_arn        = aws_acm_certificate_validation.workspace_public.certificate_arn
+  cloudfront_certificate_arn           = aws_acm_certificate_validation.public_north_virginia.certificate_arn
+  publishing_service_domain            = var.publishing_service_domain
+  workspace                            = local.workspace
+  is_default_workspace                 = local.is_default_workspace
+  allowlist_cidrs                      = local.is_default_workspace ? concat(var.office_cidrs_list, data.fastly_ip_ranges.fastly.cidr_blocks, local.aws_nat_gateways_cidrs) : concat(var.office_cidrs_list, local.aws_nat_gateways_cidrs)
+  rails_assets_s3_regional_domain_name = aws_s3_bucket.backends_rails_assets.bucket_regional_domain_name
+
+  fronted_apps = {
+    "publisher" = { security_group_id = module.publisher_web.security_group_id },
+    "signon"    = { security_group_id = module.signon.security_group_id },
+  }
+}
+
+## Publisher
+
+resource "aws_lb_target_group" "publisher" {
+  name        = "backends-origin-publisher-${local.workspace}"
+  port        = 80
+  protocol    = "HTTP"
+  vpc_id      = local.vpc_id
+  target_type = "ip"
+
+  health_check {
+    path = "/healthcheck"
+  }
+}
+
+resource "aws_lb_listener_rule" "publisher" {
+  listener_arn = module.backends_origin.origin_alb_listerner_arn
+  priority     = 1
+
+  action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.publisher.arn
+  }
+
+  condition {
+    http_header {
+      http_header_name = "X-Cloudfront-Token"
+      values           = [module.backends_origin.origin_alb_x_custom_header_secret]
+    }
+  }
+
+  condition {
+    host_header {
+      values = ["publisher.*"]
+    }
+  }
+}
+
+resource "aws_route53_record" "publisher" {
+  zone_id = aws_route53_zone.workspace_public.zone_id
+  name    = "publisher"
+  type    = "CNAME"
+  ttl     = 300
+  records = [module.backends_origin.fqdn]
+}
+
+## Signon
+
+resource "aws_lb_target_group" "signon" {
+  name        = "backends-origin-signon-${local.workspace}"
+  port        = 80
+  protocol    = "HTTP"
+  vpc_id      = local.vpc_id
+  target_type = "ip"
+
+  health_check {
+    path = "/healthcheck"
+  }
+}
+
+resource "aws_lb_listener_rule" "signon_private_api_allow_unrestricted_cidrs" {
+  listener_arn = module.backends_origin.origin_alb_listerner_arn
+  priority     = 2
+
+  action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.signon.arn
+  }
+
+  condition {
+    host_header {
+      values = ["signon.*"]
+    }
+  }
+
+  condition {
+    path_pattern {
+      values = ["/api/*"]
+    }
+  }
+
+  condition {
+    source_ip {
+      values = concat(var.office_cidrs_list, local.aws_nat_gateways_cidrs)
```

Comment:
FYI. I recall seeing an issue here when applying - this list might be over the max permitted, though I can't see a [quota for this](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-limits.html).

Reply:
Ah yes, the terraform apply is failing in your workspace:

```
Error: Error creating LB Listener Rule: ValidationError: A rule can only have '5' condition values

  on backends_origin.tf line 87, in resource "aws_lb_listener_rule" "signon_private_api_allow_unrestricted_cidrs":
  87: resource "aws_lb_listener_rule" "signon_private_api_allow_unrestricted_cidrs" {

```

Reply:
I would recommend just removing the `var.office_cidrs_list` - VPN users don't need access to this endpoint.

Reply:
I can do that.

Reply:
I implemented it in the waf in the end due to the restriction to 5 comparisons per lb rule